eugl/mndwi.py

```diff
@@ -0,0 +1,381 @@
+        )
+
+        attrs = {
+            "crs_wkt": geobox.crs.ExportToWkt(),
+            "geotransform": geobox.transform.to_gdal(),
+            "no_data_value": nodata,
+            "granule": granule,
+            "description": desc,
+            "n_lines": nRows,
+            "n_samples": nCols,
+            "platform": platform_id,
+            "spatial_resolution": spatial_res,
+        }
+
+        # add attrs to dataset
+        for key in attrs:
+            mndwi_ds.attrs[key] = attrs[key]
+
+        mask = (green_im == nodata) | (swir_im == nodata)
+        mndwi_im = np.zeros([nRows, nCols], order="C", dtype=np.float32)
+        for tile in tiles:
+            mndwi_tile = compute_mndwi(green_im[tile], swir_im[tile])
+
+            # perform masking
+            mndwi_tile[~np.isfinite(mndwi_tile)] = nodata
+            mndwi_tile[mask[tile]] = nodata
+
+            mndwi_ds[tile] = mndwi_tile
+            mndwi_im[tile] = mndwi_tile
+
+        lowerVal, UpperVal = np.percentile(mndwi_im[~mask], (1, 99))
+
+        lowerVal_arr[i] = lowerVal
+        upperVal_arr[i] = UpperVal
+
+    h5_fid.create_dataset(
+        "mndwi_1st_percentiles",
+        data=lowerVal_arr,
+        shape=lowerVal_arr.shape,
+        dtype="float32",
+        compression="lzf",
+    )
+
+    h5_fid.create_dataset(
+        "mndwi_99th_percentiles",
+        data=upperVal_arr,
+        shape=upperVal_arr.shape,
+        dtype="float32",
+        compression="lzf",
+    )
+
+    del green_im, swir_im, mndwi_im
+
+    # ----------------------------------- #
+    #  Create  an rgb image (uint8) that  #
+    #  can directly be  opened from  the  #
+    #  MNDWI H5 file. The following code  #
+    #  will be deleted after testing      #
+    # ----------------------------------- #
+
+    # get the rgb band paths from the first product
+    red_path, grn_path, blu_path = get_RGB_bands(platform_id, products[0], paths)
+    rgb_paths = [red_path, grn_path, blu_path]
+    rgb_names = ["red", "green", "blue"]
+
+    # create a group
+    rgb_grp = h5_fid.create_group("seadas_rgb")
+
+    # because no reprojection is necessary, we can load
+    # the red & blue bands as tiles to conserve memory
+    for i in range(0, len(rgb_names)):
+
+        chunks = grp[rgb_paths[i]].chunks
+
+        rgb_ds = rgb_grp.create_dataset(
+            rgb_names[i],
+            shape=(nRows, nCols),
+            dtype="uint8",
+            compression="lzf",
+            chunks=chunks,
+            shuffle=True,
+        )
+
+        # create h5 attributes
+        desc = "scaled {0} band ({1} m) derived from {2} {3}".format(
+            rgb_names[i],
+            int(spatial_res),
+            products[0],
+            psplit(rgb_paths[i])[-1],
+        )
+        attrs["description"] = desc
+
+        # add attrs to dataset
+        for key in attrs:
+            rgb_ds.attrs[key] = attrs[key]
+
+        # For some reason I have to recreate the tiles. Attempts
```

Comment:
Delete the comment.

A generator is a one life object.  If you want to keep it to loop again later convert the generator to a list. e.g.:

```python
tiles = list(generate_tiles(...))
```

Reply:
Ahhh thanks - fixed